### PR TITLE
base: Fix issues with musl libc

### PR DESCRIPTION
**Pull request by nanoslayer**
These commits fix a few issues that I encountered on Linux with musl libc.

### CODE REVIEW

**Comment by maujin111** [comment]
Could you provide the steps to reproduce the issues?

**Comment by nanoslayer** [comment]
- To trigger the segfault on pthread_detach, simply open any file. LibreSprite will crash.
- To trigger the path normalization bug, create a new file and try to save it. An error saying that `LibreSprite can't save "" files.` will appear because the path normalization failed somehow. The normalization probably should be fixed, but for now we can simply avoid it.

File: src/doc/document.cpp [hunk 1]
```diff
@@ -64,12 +64,7 @@ std::string Document::name() const
 
 void Document::setFilename(const std::string& filename)
 {
-  // Normalize the path (if the filename has a path)
```
Comment (maujin111) [inline src/doc/document.cpp:67]: add a comment for fix it later

**Comment by Zughy** [comment]
>     * The normalization probably should be fixed, but for now we can simply avoid it.

Fixing a bug introducing a new bug doesn't seem a great way to really fix things

**Comment by nanoslayer** [comment]
Alright. I've reverted the normalization workaround. I'll fix it in its own pull request.

**Comment by nanoslayer** [comment]
I've just pushed a proper fix for the path normalization error.

**Review by maujin111 — APPROVED** [review]

**Pull request merged**